src/NUnitFramework/framework/Api/FrameworkController.cs

```diff
@@ -126,18 +127,20 @@ public FrameworkController(Assembly assembly, string idPrefix, IDictionary setti
         private void Initialize(string assemblyPath, IDictionary settings)
         {
             AssemblyNameOrPath = assemblyPath;
-            Settings = settings;
 
-            if (settings.Contains(PackageSettings.InternalTraceLevel))
+            var newSettings = settings as IDictionary<string, object>;
+            Settings = newSettings ?? settings.Cast<DictionaryEntry>().ToDictionary(de => (string)de.Key, de => de.Value);
```

Comment:
I'm unsure about the need for this copying. I'll enlarge in the general comment on the PR.